### PR TITLE
Tos verify

### PR DESCRIPTION
**Pull request by zhouji3x**

### CODE REVIEW

File: libkernelflinger/trusty_common.c [hunk 1]
```diff
@@ -44,6 +44,138 @@
 #include "gpt.h"
 #include "efilinux.h"
 
+#define AVB_COMPILATION
```
Comment (GangSecurity) [inline libkernelflinger/trusty_common.c:47]: It is not used

File: libkernelflinger/trusty_common.c [hunk 2]
```diff
@@ -44,6 +44,138 @@
...
+            *image_size = pdata->data_size;
+            return EFI_SUCCESS;
+        }
+    }
+
+    *image = NULL;
+    return EFI_NOT_FOUND;
+}
+
+AvbSlotVerifyResult avb_verify_image(const uint8_t *image_buf, size_t length)
```
Comment (GangSecurity) [inline libkernelflinger/trusty_common.c:77]: static function

File: libkernelflinger/trusty_common.c (hunk 1 above)
Comment (zhouji3x) [reply libkernelflinger/trusty_common.c:47]: AVB_COMPILATION is not defined for libkernelflinger,  so it must be defined here

File: libkernelflinger/trusty_common.c (hunk 2 above)
Comment (zhouji3x) [reply libkernelflinger/trusty_common.c]: done

**Review by GangSecurity — APPROVED** [review]

File: libkernelflinger/gpt.c [hunk 3]
```diff
@@ -916,3 +916,66 @@ EFI_STATUS gpt_get_partitions(struct gpt_partition **partitions, UINTN *size, lo
 
 	return memcpy_s(*partitions, *size, sdisk.partitions, *size);
 }
+
+UINT64 get_partition_start(struct gpt_partition_interface *gparti)
+{
```
Comment (GangSecurity) [inline libkernelflinger/gpt.c:921]: check parameter

File: libkernelflinger/gpt.c [hunk 4]
```diff
@@ -916,3 +916,66 @@ EFI_STATUS gpt_get_partitions(struct gpt_partition **partitions, UINTN *size, lo
...
+	return gparti->part.starting_lba * gparti->bio->Media->BlockSize;
+}
+
+UINT64 get_partition_size(struct gpt_partition_interface *gparti)
+{
+	return (gparti->part.ending_lba + 1 - gparti->part.starting_lba) *
```
Comment (GangSecurity) [inline libkernelflinger/gpt.c:927]: check parameters before use

File: libkernelflinger/gpt.c (hunk 3 above)
Comment (zhouji3x) [reply libkernelflinger/gpt.c:921]: done

File: libkernelflinger/gpt.c (hunk 4 above)
Comment (zhouji3x) [reply libkernelflinger/gpt.c:933]: done

**Review by GangSecurity — APPROVED** [review]

**Pull request merged**